src/universal_transfer_operator/integrations/hooks/fivetran.py

```diff
@@ -0,0 +1,361 @@
+        https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/hooks/base_aws.py#L486
+        """
+        try:
+            # This can be a circular import under specific configurations.
+            # Importing locally to either avoid or catch it if it does happen.
+            from airflow import __version__ as airflow_version
+
+            return "-airflow_version/" + airflow_version
+        except Exception:
+            # Under no condition should an error here ever cause an issue for the user.
+            return ""
+
+    def __init__(
+        self,
+        fivetran_conn_id: str = "fivetran",
+        fivetran_conn=None,
+        timeout_seconds: int = 180,
+        retry_limit: int = 3,
+        retry_delay: float = 1.0,
+    ) -> None:
+        super().__init__(None)  # Passing None fixes a runtime problem in Airflow 1
+        self.conn_id = fivetran_conn_id
+        self.fivetran_conn = fivetran_conn
+        self.timeout_seconds = timeout_seconds
+        if retry_limit < 1:
+            raise ValueError("Retry limit must be greater than equal to 1")
+        self.retry_limit = retry_limit
+        self.retry_delay = retry_delay
+
+    def _do_api_call(self, endpoint_info, json=None):
+        """
+        Utility function to perform an API call with retries
+
+        :param endpoint_info: Tuple of method and endpoint
+        :param json: Parameters for this API call.
+        """
+        method, endpoint = endpoint_info
+        if self.fivetran_conn is None:
+            self.fivetran_conn = self.get_connection(self.conn_id)
+        auth = (self.fivetran_conn.login, self.fivetran_conn.password)
+        url = f"{self.api_protocol}://{self.api_host}/{endpoint}"
+
+        headers = {"User-Agent": self.api_user_agent + self._get_airflow_version()}
+
+        if method == "GET":
+            request_func = requests.get
+        elif method == "POST":
+            request_func = requests.post
+            headers.update({"Content-Type": "application/json;version=2"})
+        elif method == "PATCH":
+            request_func = requests.patch
+            headers.update({"Content-Type": "application/json;version=2"})
+        else:
+            raise AirflowException("Unexpected HTTP Method: " + method)
+
+        attempt_num = 1
+        while True:
+            try:
+                response = request_func(
+                    url,
+                    data=json if method in ("POST", "PATCH") else None,
+                    params=json if method in ("GET") else None,
+                    auth=auth,
+                    headers=headers,
+                )
+                response.raise_for_status()
+                return response.json()
+            except requests_exceptions.RequestException as e:
+                if not _retryable_error(e):
+                    # In this case, the user probably made a mistake.
+                    # Don't retry.
```

Comment:
This assumption may not be true, there can be a timeout or too many requests. 
ref - https://requests.readthedocs.io/en/latest/_modules/requests/exceptions/
and in those scenarios, it's a valid thing to retry.

Reply:
Can you add a code comment for the same?

Reply:
Created issue here: https://github.com/astronomer/apache-airflow-provider-transfers/issues/64